go/vt/vtctl/workflow/server.go

```diff
@@ -1705,6 +1705,16 @@ func (s *Server) VDiffCreate(ctx context.Context, req *vtctldatapb.VDiffCreateRe
 			req.TargetKeyspace, req.Workflow)
 	}
 
+	workflowStatus, err := s.getWorkflowStatus(ctx, req.TargetKeyspace, req.Workflow)
+	if err != nil {
+		return nil, err
+	}
+	if workflowStatus != binlogdatapb.VReplicationWorkflowState_Running {
+		log.Infof("Workflow %s.%s is not running, cannot start VDiff in state %s", req.TargetKeyspace, req.Workflow, workflowStatus)
+		return nil, vterrors.Errorf(vtrpcpb.Code_FAILED_PRECONDITION,
+			"not all streams are running in workflow %s.%s", req.TargetKeyspace, req.Workflow)
+	}
```

Comment:
I'm curious why this addition is special to Online DDL. Is this about a multi-shard Online DDL, tested across all shards at once?

Reply:
This isn't specific to OnlineDDL, it's just something that Rohit and I had discussed that we should check and prevent generally.